### PR TITLE
Canvas - add restrictions on editing relation arrows

### PR DESCRIPTION
**Pull request by mdroidian**
https://www.loom.com/share/db47211b0c6b44e5821cf7b62364de5c

### CODE REVIEW

File: src/components/TldrawCanvas.tsx [hunk 1]
```diff
@@ -1090,12 +1170,33 @@ const TldrawCanvas = ({ title }: Props) => {
                         // TODO: find a better way to identify custom arrow shapes
                         // shape.type or shape.name probably?
                         if (
-                          !allRelationIdSet.has(shape.type) ||
+                          !allRelationIdSet.has(shape.type) &&
```
Comment (mdroidian) [inline src/components/TldrawCanvas.tsx:1173]: Opa! 😁

File: src/components/TldrawCanvas.tsx [hunk 2]
```diff
@@ -1072,6 +1093,62 @@ const TldrawCanvas = ({ title }: Props) => {
...
+                    return class extends Translate {
+                      override onPointerUp: TLPointerEvent = () => {
+                        this.onComplete({
+                          type: "misc",
+                          name: "complete",
+                        });
+                        const shape = this.app.getShapeById(this.info.shape.id);
+                        if (!shape) return;
+                        // TODO: find a better way to identify custom arrow shapes
+                        // shape.type or shape.name probably?
```
Comment (dvargas92495) [inline src/components/TldrawCanvas.tsx:1111]: could be useful to hoist this logic up to a helper to consolidate logic we want to improve to one place

File: src/components/TldrawCanvas.tsx [hunk 3]
```diff
@@ -1072,6 +1093,62 @@ const TldrawCanvas = ({ title }: Props) => {
...
+                        if (
+                          !allRelationIdSet.has(shape.type) &&
+                          !allAddReferencedNodeActionsSet.has(shape.type)
+                        )
+                          return;
+
+                        // Stop accidental arrow reposition
+                        const cancelAndWarn = (content: string) => {
```
Comment (dvargas92495) [inline src/components/TldrawCanvas.tsx:1119]: this `cancelAndWarn` also seems like it could be hoisted

**Review by dvargas92495 — APPROVED** [review]
always wanted this!

**Pull request merged**